sdk/python/kfp/components/structures.py

```diff
@@ -38,39 +37,21 @@
 
 
 @dataclasses.dataclass
-class InputSpec_:
-    """Component input definitions. (Inner class).
+class InputSpec:
+    """Component input definitions.
 
     Attributes:
         type: The type of the input.
         default (optional): the default value for the input.
-        description: Optional: the user description of the input.
+        optional: Wether the input is optional. An input is optional when it has an explicit default value.
     """
     type: Union[str, dict]
     default: Optional[Any] = None
-
-
-# Hack to allow access to __init__ arguments for setting _optional value
-class InputSpec(InputSpec_):
-    """Component input definitions.
-
-    Attributes:
-        type: The type of the input.
-        default (optional): the default value for the input.
-        _optional: Wether the input is optional. An input is optional when it has an explicit default value.
-    """
-
-    @functools.wraps(InputSpec_.__init__)
-    def __init__(self, *args, **kwargs) -> None:
-        """InputSpec constructor, which can access the arguments passed to the
-        constructor for setting ._optional value."""
-        if args:
-            raise ValueError('InputSpec does not accept positional arguments.')
-        super().__init__(*args, **kwargs)
-        self._optional = 'default' in kwargs
+    optional: Optional[bool] = False
```

Comment:
nit: looks like `optional` will never be `None`? If so, we can remove `Optional[...]` in the type annotation.